In [5]:
import pandas as pd
DATA_PATH = "notebooks/book-text-ml/book_bestseller_clean.csv"
df_books = pd.read_csv(DATA_PATH, encoding="utf-8-sig")
print("데이터 크기:", df_books.shape)
print("컬럼:", df_books.columns.tolist())
print("상품명 결측치:", df_books["상품명"].isna().sum())
df_books[["상품명"]].head(10)

데이터 크기: (199, 8)
컬럼: ['순위', '판매상품ID', '상품명', '판매가', '저자', '출판사', '발행일', '분야']
상품명 결측치: 0


,상품명
0,소년이 온다
1,모순
2,결국 국민이 합니다
3,혼모노
4,급류
5,초역 부처의 말
6,청춘의 독서(특별증보판)
7,어른의 행복은 조용하다
8,채식주의자
9,단 한 번의 삶(강물에디션 활판인쇄 한정판)


### 실습 2. 분석할 제목 문자열 준비하기

Vectorizer에 전달할 제목을 문자열 형태로 준비합니다.

In [6]:
import pandas as pd

# (참고) df_books 데이터프레임이 이전 단계에서 이미 로드되어 있다고 가정합니다.
# 예: df_books = pd.read_csv("book_bestseller_clean.csv")

# 1. '상품명' 컬럼의 텍스트 데이터 전처리
titles = (
    df_books["상품명"]
    .fillna("")        # 결측치(비어있는 값)를 빈 문자열("")로 채워 오류를 방지합니다.
    .astype(str)       # 모든 데이터를 안전하게 문자열(String) 타입으로 변환합니다.
    .str.strip()       # 문자열 앞뒤에 있는 불필요한 공백을 깔끔하게 제거합니다.
)

# 2. 전처리 후 빈 문자열("")이 되어버린 데이터를 제외하고, 인덱스를 처음부터 다시 맞춥니다.
titles = titles[titles != ""].reset_index(drop=True)

# 3. 최종적으로 분석에 사용할 제목의 개수를 출력하여 확인합니다.
print("사용할 제목 수:", len(titles))

# 4. 결과가 잘 담겼는지 상위 10개의 데이터를 확인합니다.
titles.head(10)

사용할 제목 수: 199


0                      소년이 온다
1                          모순
2                  결국 국민이 합니다
3                         혼모노
4                          급류
5                    초역 부처의 말
6               청춘의 독서(특별증보판)
7                어른의 행복은 조용하다
8                       채식주의자
9    단 한 번의 삶(강물에디션 활판인쇄 한정판)
Name: 상품명, dtype: str

### 실행 결과 요약

1. **사용할 제목 수**
   - 전처리를 거쳐 최종적으로 분석에 사용될 도서 제목의 수는 총 199개입니다[cite: 2].

2. **결과 데이터 샘플 (상위 10개)**
   - 0: 소년이 온다[cite: 2]
   - 1: 모순[cite: 2]
   - 2: 결국 국민이 합니다[cite: 2]
   - 3: 혼모노[cite: 2]
   - 4: 급류[cite: 2]
   - 5: 초역 부처의 말[cite: 2]
   - 6: 청춘의 독서(특별증보판)[cite: 2]
   - 7: 어른의 행복은 조용하다[cite: 2]
   - 8: 채식주의자[cite: 2]
   - 9: 단 한 번의 삶(강물에디션 활판인쇄 한정판)[cite: 2]

3. **데이터 속성**
   - 시리즈 이름(Name): 상품명[cite: 2]
   - 데이터 타입(dtype): 문자열(str)[cite: 2]